pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java

```diff
@@ -89,35 +107,39 @@ public void reduceAndSetResults(String tableName, DataSchema dataSchema, Map<Ser
     int resultSize = 0;
     Collection<DataTable> dataTables = dataTableMap.values();
 
-    // Aggregation group-by query.
-    // read results as records if  GROUP_BY_MODE is explicitly set to SQL
-
     if (_groupByModeSql) {
 
-      // if RESPONSE_FORMAT is SQL, return results in {@link ResultTable}
       if (_responseFormatSql) {
-        setSQLGroupByOrderByResults(brokerResponseNative, dataSchema, _brokerRequest.getAggregationsInfo(),
-            _brokerRequest.getGroupBy(), _brokerRequest.getOrderBy(), dataTables);
+        setSQLGroupByInResultTable(brokerResponseNative, dataSchema, dataTables);
         resultSize = brokerResponseNative.getResultTable().getRows().size();
       } else {
-        setPQLGroupByOrderByResults(brokerResponseNative, dataSchema, _brokerRequest.getAggregationsInfo(),
-            _brokerRequest.getGroupBy(), _brokerRequest.getOrderBy(), dataTables);
+        setSQLGroupByInAggregationResults(brokerResponseNative, dataSchema, dataTables);
         if (!brokerResponseNative.getAggregationResults().isEmpty()) {
           resultSize = brokerResponseNative.getAggregationResults().get(0).getGroupByResult().size();
         }
       }
     } else {
 
-      boolean[] aggregationFunctionSelectStatus =
-          AggregationFunctionUtils.getAggregationFunctionsSelectStatus(_brokerRequest.getAggregationsInfo());
-      setGroupByHavingResults(brokerResponseNative, aggregationFunctionSelectStatus, _brokerRequest.getGroupBy(),
-          dataTables, _brokerRequest.getHavingFilterQuery(), _brokerRequest.getHavingFilterSubQueryMap());
-      // We emit the group by size when the result isn't empty. All the sizes among group-by results should be the same.
-      // Thus, we can just emit the one from the 1st result.
-      if (!brokerResponseNative.getAggregationResults().isEmpty()) {
-        resultSize = brokerResponseNative.getAggregationResults().get(0).getGroupByResult().size();
+      if (_responseFormatSql && _numAggregationFunctions > 1) {
+        brokerResponseNative.addToExceptions(new QueryProcessingException(QueryException.MERGE_RESPONSE_ERROR_CODE,
```

Comment:
Are we throwing this exception after all of query execution? Could we catch this upfront so we don't waste resources?

Reply:
Good point. We could check for this in `CombineGroupByOperator`, and set exception and return. Does that sound like a good place?

Reply:
Actually, even that is too late, as we would be wasting segment level resources.

Reply:
Added this check to validateRequest in BaseBrokerRequestHandler, which happens before all the execution

Reply:
Would be good to add a test for the same in QueryValidationTest -- it already has some tests for invalid queries that call into BaseBrokerRequestHandler.validateRequest()

Reply:
Removed this check and exception